### ISSUE TITLE
Merging new PRs first then old ones later sometimes messes up the release PR checkboxes

### ISSUE THREAD

**Issue by hkurokawa** [motemen/git-pr-release#20]
# Abstract
In my production, I sometimes merge the latest PR and then merge an older PR. I don't know why but sometimes it messes up the bullet list of PRs in the release PR. Some PRs to be released are missing and some are duplicated. Unfortunately, I could not reproduce the issue with a simple example so I believe it deeply depends on the content of the release PR.

# Steps to reproduce
Please note the steps do not always reproduce the issue as I noted above. However, in my experience, the similar steps often lead to the issue.

1. Open a PR A
2. Open another PR B
3. Merge PR B
4. Open and merge PR C
5. Run `git-pr-release` to generate the release PR
6. Merge PR A
7. Run `git-pr-release` again to refresh the release PR

# Result
After step 6, I sometimes see PR A is missing or PR C is duplicated.

# Expected result
After step 6, the release PR does not contain any missing PR (A, B, C are not missing) and does not contain any duplicated PRs.

# Logs
Here is a log of `DBUG=true git-pr-release -n` with some sensitive information masked.

<details>
<code>
<pre>
% DEBUG=true git-pr-release -n
Executing `git config remote.origin.url`
Executing `git config -f .git-pr-release pr-release.branch.production`
Executing `git config pr-release.branch.production`
Executing `git config -f .git-pr-release pr-release.branch.staging`
Repository:        hkurokawa/git-pr-release
Production branch: master
Staging branch:    develop
Executing `git config -f .git-pr-release pr-release.token`
Executing `git config pr-release.token`
Executing `git remote update origin`
Executing `git log --merges --pretty=format:%P origin/master..origin/develop`
Executing `git ls-remote origin refs/pull/*/head`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
#7180 (xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx) is already merged into master
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
Executing `git merge-base xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx origin/master`
To be released: #6939 xxxxxxxxxx
To be released: #6940 xxxxxxxxxx
To be released: #6941 xxxxxxxxxx
To be released: #7024 xxxxxxxxxx
To be released: #7043 xxxxxxxxxx
To be released: #7081 xxxxxxxxxx
To be released: #7104 xxxxxxxxxx
To be released: #7142 xxxxxxxxxx
To be released: #7164 xxxxxxxxxx
To be released: #7178 xxxxxxxxxx
To be released: #7186 xxxxxxxxxx
To be released: #7192 xxxxxxxxxx
To be released: #7193 xxxxxxxxxx
To be released: #7196 xxxxxxxxxx
To be released: #7198 xxxxxxxxxx
To be released: #7204 xxxxxxxxxx
To be released: #7206 xxxxxxxxxx
To be released: #7207 xxxxxxxxxx
To be released: #7209 xxxxxxxxxx
To be released: #7211 xxxxxxxxxx
To be released: #7212 xxxxxxxxxx
To be released: #7216 XXXXXXXXXX
To be released: #7217 xxxxxxxxxx
To be released: #7219 xxxxxxxxxx
To be released: #7222 xxxxxxxxxx
To be released: #7228 xxxxxxxxxx
To be released: #7231 xxxxxxxxxx
To be released: #7234 xxxxxxxxxx
Searching for existing release pull requests...
Executing `git config -f .git-pr-release pr-release.template`
Executing `git rev-parse --show-toplevel`
diff: ["!", [0, "- [x] #6939 xxxxxxxxxx @hkurokawa"], [0, "- [ ] #6939 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #6939 xxxxxxxxxx @hkurokawa
diff: ["!", [1, "- [x] #6940 xxxxxxxxxx @hkurokawa"], [1, "- [ ] #6940 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #6940 xxxxxxxxxx @hkurokawa
diff: ["!", [2, "- [x] #6941 xxxxxxxxxx @hkurokawa"], [2, "- [ ] #6941 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #6941 xxxxxxxxxx @hkurokawa
diff: ["!", [3, "- [x] #7024 xxxxxxxxxx @hkurokawa"], [3, "- [ ] #7024 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7024 xxxxxxxxxx @hkurokawa
diff: ["!", [4, "- [x] #7043 xxxxxxxxxx @hkurokawa"], [4, "- [ ] #7043 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7043 xxxxxxxxxx @hkurokawa
diff: ["!", [5, "- [x] #7081 xxxxxxxxxx @hkurokawa"], [5, "- [ ] #7081 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7081 xxxxxxxxxx @hkurokawa
diff: ["!", [6, "- [x] #7142 xxxxxxxxxx @hkurokawa"], [6, "- [ ] #7104 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7142 xxxxxxxxxx @hkurokawa
diff: ["!", [7, "- [x] #7164 xxxxxxxxxx @hkurokawa"], [7, "- [ ] #7142 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7164 xxxxxxxxxx @hkurokawa
diff: ["!", [8, "- [x] #7178 xxxxxxxxxx @hkurokawa"], [8, "- [ ] #7164 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7178 xxxxxxxxxx @hkurokawa
diff: ["!", [9, "- [x] #7186 xxxxxxxxxx @hkurokawa"], [9, "- [ ] #7178 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7186 xxxxxxxxxx @hkurokawa
diff: ["!", [10, "- [x] #7192 xxxxxxxxxx @hkurokawa"], [10, "- [ ] #7186 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7192 xxxxxxxxxx @hkurokawa
diff: ["!", [11, "- [x] #7193 xxxxxxxxxx @hkurokawa"], [11, "- [ ] #7192 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7193 xxxxxxxxxx @hkurokawa
diff: ["!", [12, "- [x] #7196 xxxxxxxxxx @hkurokawa"], [12, "- [ ] #7193 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7196 xxxxxxxxxx @hkurokawa
diff: ["!", [13, "- [x] #7198 xxxxxxxxxx @hkurokawa"], [13, "- [ ] #7196 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7198 xxxxxxxxxx @hkurokawa
diff: ["!", [14, "- [x] #7204 xxxxxxxxxx @hkurokawa"], [14, "- [ ] #7198 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7204 xxxxxxxxxx @hkurokawa
diff: ["!", [15, "- [x] #7206 xxxxxxxxxx @hkurokawa"], [15, "- [ ] #7204 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7206 xxxxxxxxxx @hkurokawa
diff: ["!", [16, "- [x] #7207 xxxxxxxxxx @hkurokawa"], [16, "- [ ] #7206 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7207 xxxxxxxxxx @hkurokawa
diff: ["!", [17, "- [x] #7209 xxxxxxxxxx @hkurokawa"], [17, "- [ ] #7207 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7209 xxxxxxxxxx @hkurokawa
diff: ["!", [18, "- [x] #7211 xxxxxxxxxx @hkurokawa"], [18, "- [ ] #7209 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7211 xxxxxxxxxx @hkurokawa
diff: ["!", [19, "- [x] #7212 xxxxxxxxxx @hkurokawa"], [19, "- [ ] #7211 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7212 xxxxxxxxxx @hkurokawa
diff: ["!", [20, "- [x] #7216 xxxxxxxxxx @hkurokawa"], [20, "- [ ] #7212 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7216 xxxxxxxxxx @hkurokawa
diff: ["!", [21, "- [x] #7217 xxxxxxxxxx @hkurokawa"], [21, "- [ ] #7216 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7217 xxxxxxxxxx @hkurokawa
diff: ["!", [22, "- [x] #7219 xxxxxxxxxx @hkurokawa"], [22, "- [ ] #7217 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7219 xxxxxxxxxx @hkurokawa
diff: ["!", [23, "- [x] #7222 xxxxxxxxxx @hkurokawa"], [23, "- [ ] #7219 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7222 xxxxxxxxxx @hkurokawa
diff: ["!", [24, "- [x] #7228 xxxxxxxxxx @hkurokawa"], [24, "- [ ] #7222 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7228 xxxxxxxxxx @hkurokawa
diff: ["+", [25, ""], [25, "- [ ] #7228 xxxxxxxxxx @hkurokawa"]]
Use line as is: - [ ] #7228 xxxxxxxxxx @hkurokawa"+", [25, ""], [26, "- [ ] #7231 xxxxxxxxxx @hkurokawa"]]
Use line as is: - [ ] #7231 xxxxxxxxxx @hkurokawa"+", [25, ""], [27, "- [ ] #7234 xxxxxxxxxx @hkurokawa"]]
Use line as is: - [ ] #7234 xxxxxxxxxx @hkurokawa
diff: ["=", [25, ""], [28, ""]]
Use line as is:
diff: ["=", [26, "## リリース作業"], [29, "## リリース作業"]]
Use line as is: ## リリース作業
diff: ["=", [27, ""], [30, ""]]
Use line as is:
diff: ["=", [28, "- [ ] apk作成(production/staging release)"], [31, "- [ ] apk作成(production/staging release)"]]
Use line as is: - [ ] apk作成(production/staging release)
Dry-run. Not updating PR
Release 2017-12-29 12:15:28 +0900
- [x] #6939 xxxxxxxxxx @hkurokawa
- [x] #6940 xxxxxxxxxx @hkurokawa
- [x] #6941 xxxxxxxxxx @hkurokawa
- [x] #7024 xxxxxxxxxx @hkurokawa
- [x] #7043 xxxxxxxxxx @hkurokawa
- [x] #7081 xxxxxxxxxx @hkurokawa
- [x] #7142 xxxxxxxxxx @hkurokawa
- [x] #7164 xxxxxxxxxx @hkurokawa
- [x] #7178 xxxxxxxxxx @hkurokawa
- [x] #7186 xxxxxxxxxx @hkurokawa
- [x] #7192 xxxxxxxxxx @hkurokawa
- [x] #7193 xxxxxxxxxx @hkurokawa
- [x] #7196 xxxxxxxxxx @hkurokawa
- [x] #7198 xxxxxxxxxx @hkurokawa
- [x] #7204 xxxxxxxxxx @hkurokawa
- [x] #7206 xxxxxxxxxx @hkurokawa
- [x] #7207 xxxxxxxxxx @hkurokawa
- [x] #7209 xxxxxxxxxx @hkurokawa
- [x] #7211 xxxxxxxxxx @hkurokawa
- [x] #7212 xxxxxxxxxx @hkurokawa
- [x] #7216 xxxxxxxxxx @hkurokawa
- [x] #7217 xxxxxxxxxx @hkurokawa
- [x] #7219 xxxxxxxxxx @hkurokawa
- [x] #7222 xxxxxxxxxx @hkurokawa
- [x] #7228 xxxxxxxxxx @hkurokawa
- [ ] #7228 xxxxxxxxxx @hkurokawa
- [ ] #7231 xxxxxxxxxx @hkurokawa
- [ ] #7234 xxxxxxxxxx @hkurokawa

## リリース作業

- [ ] apk作成(production/staging release)
</pre>
</code>
</details>

As you can see, the result is missing `#7104` and `#7228` is duplicated.

It seems to me the diff command guessed `#7142` was replaced with `#7104` which is not true and omitted `#7104`.

```
diff: ["!", [6, "- [x] #7142 xxxxxxxxxx @hkurokawa"], [6, "- [ ] #7104 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7142 xxxxxxxxxx @hkurokawa
```

Thus, the following diffs were all wrong and this led to the second issue, `#7228` is duplicated.

```
diff: ["!", [24, "- [x] #7228 xxxxxxxxxx @hkurokawa"], [24, "- [ ] #7222 xxxxxxxxxx @hkurokawa"]]
Found checklist diff; use old one: - [x] #7228 xxxxxxxxxx @hkurokawa
diff: ["+", [25, ""], [25, "- [ ] #7228 xxxxxxxxxx @hkurokawa"]]
Use line as is: - [ ] #7228 xxxxxxxxxx @hkurokawa"+", [25, ""], [26, "- [ ] #7231 xxxxxxxxxx @hkurokawa"]]
```

If you need more information, feel free to ask me.
